components/list/list-item.js

```diff
@@ -0,0 +1,183 @@
+				display: flex;
+				flex-grow: 0;
+				flex-shrink: 0;
+				margin-right: 0.9rem;
+				max-height: 52px;
+				max-width: 90px;
+				overflow:hidden;
+			}
+			:host([dir="rtl"]) ::slotted([slot|="illustration"]) {
+				margin-left: 0.9rem;
+				margin-right: 0rem;
+			}
+			::slotted([slot="illustration-outer"]) {
+				margin-bottom: 18px;
+				margin-top: 18px;
+			}
+			::slotted([slot="actions"]) {
+				align-self: flex-start;
+				display: flex;
+				flex-grow: 0;
+			}
+			.d2l-list-item-main {
+				width: 100%;
+			}
+		`;
+		const mainContent = css`
+			::slotted(.d2l-list-item-text) {
+				margin: 0;
+				max-height: 2.4rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive),
+			::slotted(.d2l-list-item-text-secondary) {
+				margin: 0;
+				margin-top: 0.3rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: none;
+			}
+		`;
+		const divider = css`
+			.d2l-list-item-divider-top,
+			.d2l-list-item-divider-bottom {
+				border: 0;
+				border-top: 1px solid var(--d2l-color-mica);
+				margin: 0;
+				position: absolute;
+				width: 100%;
+			}
+			.d2l-list-item-divider-top {
+				display: var(--d2l-list-item-divider-top, none);
+				top: -1px;
+			}
+			.d2l-list-item-divider-bottom {
+				bottom: -1px;
+				display: var(--d2l-list-item-divider-bottom, none);
+			}
+		`;
+
+		const breakPoint580 = css`
+			.d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 71px;
+				max-width: 120px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="580"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint636 = css`
+			.d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 102px;
+				max-width: 180px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="636"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint842 = css`
+			.d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 120px;
+				max-width: 216px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="842"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+		return [ layout, mainContent, divider, breakPoint580, breakPoint636, breakPoint842];
+	}
+
+	constructor() {
+		super();
+		this._breakpoint = 0;
+		this._breakpointList = [842, 636, 580, 0];
```

Comment:
These are breakpoints I used in discover. Designers probably should pick these better.

Reply:
So... I don't love having any hardcoded breakpoints in this component. Breakpoints should be totally based on the content and how it fits in the larger page/application, and this component simply has no idea about either of those things.

For example, maybe this list is inside a 3-panel layout and therefore has a very small width. In this case, the breakpoint they'll want to use will be much smaller than if it took up the full width of the page. I think we should be doing nothing here and leave the showing/hiding totally up to the consumer.

Reply:
These breakpoints consider that though. Since it is based of the size of this component not the page.

I am thinking to make these breakpoints programmable though because it would be a better design but these could be the defaults.

Reply:
The idea behind these breakpoints is we decide how the list should show at different component widths. So when they add the list to their page it will always look correct and good. 

I like making it so the consumer doesn't have to care about responsive behaviour and can put a component any where on the screen and it will auto adjust based on it's given space.

Like for learning paths, it will have big images when it is the full page in the learning path list. But when I add this component to the add activity which is more of a side bar it will automatically look good there.

Reply:
It's a good idea in theory, but I've found when people start putting their own "stuff" in your components, the best laid plans start to fall apart. Let's give it a go though and see how it turns out!